.github/workflows/main.yml

```diff
@@ -92,9 +92,16 @@ jobs:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 50
-      - name: Start PostgreSQL
+      - name: Start PostgreSQL Ubuntu
+        if: ${{ matrix.os == 'ubuntu-latest' }}
         working-directory: docker/postgres-server
         run: docker-compose up -d && docker-compose logs
+      - name: Start PostgreSQL Windows
+        if: ${{ matrix.os == 'windows-latest' }}
+        shell: wsl-bash {0}
```

Comment:
should we extract `shell` parameter to the matrix, so we use the same step without adding extra `if`?

Reply:
Will it spawn Windows-based PostgreSQL or Linux-based one?

Reply:
Ah that's interesting. So this would be testing some JDK running on Windows vs a Linux Postgres running in Docker?

If we're going to test against Windows (i.e. duplicate what's in AppVeyor), could have a separate branch for the server itself running on Windows. The GitHub Runner installed software list for "windows-2022" says that it already has PostgreSQL 14.x installed: https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#databases

I think we'd be kind of tied to whatever they include in the runner image, but that's not any worse than what we have in AppVeyor. With the advantage that it'd work for forks too.

Reply:
> Will it spawn Windows-based PostgreSQL or Linux-based one?

Well the reason I did this was because there are no windows postgresql docker images, hence running on ubuntu

Reply:
> Ah that's interesting. So this would be testing some JDK running on Windows vs a Linux Postgres running in Docker?
> 
> If we're going to test against Windows (i.e. duplicate what's in AppVeyor), could have a separate branch for the server itself running on Windows. The GitHub Runner installed software list for "windows-2022" says that it already has PostgreSQL 14.x installed: https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#databases
> 
> I think we'd be kind of tied to whatever they include in the runner image, but that's not any worse than what we have in AppVeyor. With the advantage that it'd work for forks too.

Yeah, well ideally we want to be able to test windows with different PostgreSQL versions. If we just want v14 then we can use whatever is there